clients/drcachesim/scheduler/scheduler_impl.cpp

```diff
@@ -2894,6 +2912,22 @@ scheduler_impl_tmpl_t<RecordType, ReaderType>::update_next_record(output_ordinal
         // sched_type_t::MAP_AS_PREVIOUSLY).
         return;
     }
+    // We modify the tid and pid fields to ensure uniqueness for
+    // core-sharded-on-disk and with analyzers not using our workload identifiers.
+    // To maintain the original values, we write the workload ordinal into the top
+    // 32 bits.  We don't support distinguishing for 32-bit-build record_filter.
+    // We also ignore complexities on Mac with its 64-bit tid type.
+    int64_t workload = get_workload_ordinal(output);
+    memref_tid_t cur_tid;
+    if (record_type_has_tid(record, cur_tid) && workload > 0) {
```

Comment:
Could we add this logic and its inverse to memref.h? Since we expect users to extract the workload id and tid on their own for csod traces.

Reply:
Added the bit shift constant and helpers for extraction to memref.h.